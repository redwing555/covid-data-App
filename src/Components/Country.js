import React from 'react';
import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import countriesNames from './countriesNames';
import '../styles/country.css';

const Country = ({ country, totalConfirmed, handleClick }) => {
  const history = useHistory();
  const handleRoute = (route) => {
    history.push(route);
  };
  const isoName = countriesNames[country];

  const mapImg = isoName ? `https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/${isoName.toLowerCase()}/vector.svg` : '/World_map.png';
  return (
    <button
      className="country"
      type="button"
      onClick={() => {
        handleRoute('/country');
        handleClick(country);
      }}
    >
      <img alt="map" src={mapImg} width="100" height="100" />
      <div className="country-data">
        <span className="country-name">{country}</span>
        <span className="total-confirmed">{Intl.NumberFormat('de-DE').format(totalConfirmed)}</span>
      </div>
    </button>
  );
};

Country.propTypes = {
  country: PropTypes.string.isRequired,
  totalConfirmed: PropTypes.number.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default Country;
