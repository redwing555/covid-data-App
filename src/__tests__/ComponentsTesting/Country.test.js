import React from 'react';
import renderer from 'react-test-renderer';
import Country from '../../Components/Country';

describe('Country', () => {
  it('renders ! ', () => {
    const tree = renderer
      .create(
        <Country
          country="Morocco"
          totalConfirmed={5}
          handleClick={() => 7 + 5}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
