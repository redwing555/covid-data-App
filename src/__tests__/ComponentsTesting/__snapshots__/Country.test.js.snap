// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country renders !  1`] = `
<button
  className="country"
  onClick={[Function]}
  type="button"
>
  <img
    alt="map"
    className="country-map"
    height="100"
    src="https://raw.githubusercontent.com/djaiss/mapsicon/33ba28808f8d32b5bae0ffada9cadd07073852e1/all/ma/vector.svg"
    width="100"
  />
  <div
    className="country-data"
  >
    <svg
      className="info-icon"
      color="white"
      fill="currentColor"
      height={20}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "white",
        }
      }
      viewBox="0 0 24 24"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 18c-4.411 0-8-3.589-8-8s3.589-8 8-8 8 3.589 8 8-3.589 8-8 8z"
      />
      <path
        d="M9.293 7.707 13.586 12l-4.293 4.293 1.414 1.414L16.414 12l-5.707-5.707z"
      />
    </svg>
    <h1
      className="country-name"
    >
      MOROCCO
    </h1>
    <span
      className="total-confirmed"
    >
      5
    </span>
  </div>
</button>
`;
